### PR TITLE
tweak part invalidation logic

### PR DESCRIPTION
**Pull request by njbotkin**
Fixes https://github.com/sveltejs/sapper/issues/835.

### CODE REVIEW

**Comment by njbotkin** [comment]
Oh, hmm.  It seems that the tests added in https://github.com/sveltejs/sapper/pull/691 actually enforce the overeager invalidation behaviour described in https://github.com/sveltejs/sapper/issues/835.  I would argue that `h2` should *not* be updated in these cases (https://github.com/sveltejs/sapper/blob/master/test/apps/basics/test.ts#L292-L308).

File: runtime/src/app/app.ts [hunk 1]
```diff
@@ -262,7 +262,7 @@ function part_changed(i, segment, match, stringified_query) {
 	if (!previous) return false;
 	if (segment !== previous.segment) return true;
 	if (previous.match) {
-		if (JSON.stringify(previous.match.slice(1, i + 2)) !== JSON.stringify(match.slice(1, i + 2))) {
+		if (previous.match[0].split('/').slice(i+1, i+2)[0] !== match[0].split('/').slice(i+1, i+2)[0]) {
```
Comment (benmccann) [inline runtime/src/app/app.ts:265]: a space should be maintained on either side of the `+`. we've been playing around with setting up `eslint` to enforce style rules

**Comment by benmccann** [comment]
I'm not sure if you have any interest in pursuing this PR still? I hadn't heard anything for awhile. We've moved our CI from Travis to Github, so it'd need to be rebased at a minimum. But maybe better just to try to make sure it works in SvelteKit at this point?

**Comment by benmccann** [comment]
SvelteKit 1.0 is now out and Sapper is deprecated, so I'm going to close this

**Pull request closed without merging**